File: pset6/cash.py
from cs50 import get_float

# request and input validation
while True:
    changeFl = get_float("Change owed: ")
    if changeFl > 0:
        break

centFl = changeFl * 100     # transfer to cents
centInt = round(centFl)     # convert number to integer
coinСounter = 0     # initialize the counter

# count the number of coins for change
if centInt >= 25:
    coinСounter = centInt // 25
    centInt = centInt % 25

if centInt < 25 and centInt >= 10:
    coinСounter = coinСounter + centInt // 10
    centInt = centInt % 10

if centInt < 10 and centInt >= 5:
    coinСounter = coinСounter + centInt // 5
    centInt = centInt % 5

if centInt < 5 and centInt >= 1:
    coinСounter = coinСounter + centInt

print(coinСounter)

File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float changeFl;
    changeFl = get_float("Change owed: ");
    printf("changeFl = %f\n", changeFl);
    while (changeFl <= 0)
    {
        changeFl = get_float("Change owed: ");
    }
    float centFl = changeFl * 100;
    printf("centFl = %f\n", centFl);

    // change in cents
    // round to avoid inaccuracies
    int centInt = round(centFl);

    printf("centInt = %i\n", centInt);

    // coin counter
    int coinСounter = 0;

    // I don't understand why I need to add 1
    int remainder = centInt;
    printf("all remainder = %i, coinСounter = %i\n", remainder, coinСounter);

    while (remainder >= 25)
    {
        remainder = remainder - 25;
        coinСounter = coinСounter + 1;
        printf("25 remainder = %i, coinСounter = %i\n", remainder, coinСounter);
    }
    while (remainder < 25 && remainder >= 10)
    {
        remainder = remainder - 10;
        coinСounter = coinСounter + 1;
        printf("10 remainder = %i, coinСounter = %i\n", remainder, coinСounter);
    }
    while (remainder < 10 && remainder >= 5)
    {
        remainder = remainder - 5;
        coinСounter = coinСounter + 1;
        printf("5 remainder = %i, coinСounter = %i\n", remainder, coinСounter);
    }
    while (remainder < 5 && remainder >= 1)
    {
        remainder = remainder - 1;
        coinСounter = coinСounter + 1;
        printf("1 remainder = %i, coinСounter = %i\n", remainder, coinСounter);
    }
    printf("%i\n", coinСounter);
}